node/test/DeviceAccessorTests.ts

```diff
@@ -1,68 +1,97 @@
 // Tests for the DeviceAccessor class
 // using AVA test runner from https://github.com/avajs/ava
 
+import * as path from "path";
 import test from "ava";
 import { TestContext } from "ava";
 import { Schema } from "jsonschema";
 
 import {
     DeviceAccessor,
+    DeviceInterface,
     ITranslator,
 } from "../lib";
 
-// Move working directory from /build/test back to /test.
-process.chdir("../../test");
+// Adjust for a path that is relative to the /test directory.
+function testPath(modulePath: string): any {
+    return path.join("../../test", modulePath);
+}
 
-const deviceInterfaceA = "org.opent2t.test.A";
-const deviceInterfaceB = "org.opent2t.test.B";
+test("Load interface A", async t => {
+    let deviceInterfaceA: DeviceInterface = await
+            DeviceAccessor.getInterfaceAsync(testPath("./@opent2t/test-a"), "InterfaceA");
+    t.is(typeof deviceInterfaceA, "object") && t.truthy(deviceInterfaceA);
+    t.is(deviceInterfaceA.name, "InterfaceA");
+    t.true(Array.isArray(deviceInterfaceA.properties) && deviceInterfaceA.properties.length > 0);
+    t.true(Array.isArray(deviceInterfaceA.methods) && deviceInterfaceA.methods.length > 0);
+
+    // Extended testing on interface loading should be done by the converter tests.
+});
+
+test("Load interface B", async t => {
+    let deviceInterfaceB: DeviceInterface = await
+            DeviceAccessor.getInterfaceAsync(testPath("./@opent2t/test-b"), "InterfaceB");
+    t.is(typeof deviceInterfaceB, "object") && t.truthy(deviceInterfaceB);
+    t.is(deviceInterfaceB.name, "InterfaceB");
+    t.true(Array.isArray(deviceInterfaceB.properties) && deviceInterfaceB.properties.length > 0);
+    t.true(Array.isArray(deviceInterfaceB.methods) && deviceInterfaceB.methods.length > 0);
+
+    // Extended testing on interface loading should be done by the converter tests.
+});
 
 test("Device A property get", async t => {
-    let translatorA: ITranslator = require("../../test/translators/tA.js");
-    let deviceA = translatorA.createDevice({});
-    let propA1Value = await DeviceAccessor.getPropertyAsync(deviceA, deviceInterfaceA, "propA1");
+    let deviceA: ITranslator = await DeviceAccessor.createTranslatorAsync(
+            testPath("./@opent2t/test-a"), "TranslatorA", {});
+    t.is(typeof deviceA, "object") && t.truthy(deviceA);
+    let propA1Value = await DeviceAccessor.getPropertyAsync(deviceA, "InterfaceA", "propA1");
```

Comment:
Not quite. Translators may implement interface properties directly as JS properties, or as sync get/set methods, or async get/set methods that return promises. The `DeviceAccessor` class auto-detects which of those are available, and also throws helpful error messages when something isn't found. The reason for the flexibility is to optimize for making translators as easy to implement as possible.

---
In reply to: [73294634](https://github.com/opent2t/opent2t/pull/2#discussion_r73294634) [](ancestors = 73294634)